django/contrib/admindocs/utils.py

```diff
@@ -142,3 +142,96 @@ def default_reference_role(name, rawtext, text, lineno, inliner, options=None, c
 
     for name, urlbase in ROLES.items():
         create_reference_role(name, urlbase)
+
+named_group_beg_matcher = re.compile('\(\?P(<\w+>)')  # Matches the beginning of a named group
```

Comment:
You need to use raw strings here and in the docstrings that contain the patterns to avoid deprecation warnings for invalid escape sequences on Python 3.6 (http://bugs.python.org/issue27364).